### ISSUE TITLE
Missing match arms should not be reported on matches missing a body

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#15954]
Or rather, we shouldn't do exhaustiveness checking for matches that are missing a body, they may hide the syntax error / make it less noticable.
Example that should not trigger exhaustiveness checking
```rs
match 0;
```
